jOOR-java-8/src/test/java/org/joor/test/CompileOptionsTest.java

```diff
@@ -77,6 +77,27 @@ public void testCompileWithAnnotationProcessors() throws Exception {
         assertTrue(p.processed);
     }
 
+    @Test
+    public void testCompileWithExtraOptions() {
+        Class<?> source7Class = Reflect.compile(
+            "org.joor.test.Source7",
+            "package org.joor.test; "
+                + "public class Source7 {"
+                + "}",
+            new CompileOptions().extraOptions("-source", "7")).create().get().getClass();
+
+        // todo: check .class version, or create better test for extraOptions
+
+        Class<?> source8Class = Reflect.compile(
+            "org.joor.test.Source8",
+            "package org.joor.test; "
+                + "public class Source8 {"
+                + "}",
+            new CompileOptions().extraOptions("-source", "8")).create().get().getClass();
+
+        // todo: check .class version, or create better test for extraOptions
```

Comment:
We can implement a test that uses syntax that wasn't yet available in a given Java version. For example, default methods in Java 7, or var in Java 8